### PR TITLE
 Implement Add Current User Functionality and Forms

### PR DESCRIPTION
**Pull request by Danfall369**
## [Pull Request] Implement Project Requirements and Add Current User Functionality

Hey there!

I've implemented the project requirements and added the `current_user` functionality to the Blog App:

1️⃣ **Current User Functionality:**

- Created a `current_user` method in ApplicationController to make current user data available to all controllers.
- The method returns the first user from the database (temporary implementation, authentication to be added later).

2️⃣ **Form for Creating a New Post:**

- Added a new form for users to create posts on behalf of the current user.
- The form associates the post with the current user using the `current_user` method.

3️⃣ **Form for Creating a New Comment:**

- Included a form for users to create comments on posts on behalf of the current user.
- The form associates the comment with the current user using the `current_user` method.

4️⃣ **Adding Likes to Posts:**

- Created a new Likes Controller to handle adding likes to posts.
- When a user likes a post, the `likes_counter` for that post is incremented.

5️⃣ **View Updates:**

- Made necessary updates to the views to accommodate the new functionality and forms.
- Users can now interact seamlessly with the app, creating posts, comments, and adding likes.
- 
image
- 
image
- 
image
-
image

Please review the changes and let me know if there are any questions, feedback, or further assistance needed.

Thank you!

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+  def create
+    @post = User.find(params[:user_id]).posts.build(post_params)
+    if (@post
+        .flash[:success] = 'Post saved successfully')
+      redirect_to user_posts_path(current_user)
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+      render :new
+    end
+  end
```
Comment (franclobo) [inline app/controllers/posts_controller.rb:31]: - [ ] There is an issue with the flash method, please refactor the create method to correct this error for a better UX 👇🏽 

image

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi  @Danfall369, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Danfall369) [reply app/controllers/posts_controller.rb]: @franclobo Done, notifications now works

**Review by TadesseDev — APPROVED** [review]
### Hi @Danfall369 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**